docs: Add an explanation of using ignoring differences for app of apps pattern (#18945)
argoproj/argo-cd: docs/operator-manual/cluster-bootstrapping.md

  - resources-finalizer.argocd.argoproj.io
spec:
 ...
```

> ### Ignoring some differences in managed applications

Reviewer: Seems like this would make more sense in docs/user-guide/diffing.md. Perhaps in an example section that could contain known patterns.

What do you think?

Author: I'd like to keep some reference here, since the page is dedicated to app of apps and that's where people would come to when searching for app of apps. They'd probably not go to diffing page right away. The filters are a bit specific for app of apps, so it seems to make more sense to keep them here. Though I can potentially add changes to the diffing page and link them from this page.

Reviewer 2: @agaudreault if you Google "Argo CD app of apps" this page is the first result and is essentially the canonical source for app of apps. I think we should leave changing that pattern to another PR. In the meantime, for this PR @andrii-korotkov-verkada lets just add a link from diffing customization page to this one to discuss app of apps.

If you don't want your app of apps becoming out of sync on a regular basis and/or want to manually modify some configurations for managed applications like auto sync being enabled, you can configure ignoring differences in the app of apps, e.g.

```yaml
spec:
  ...
  syncPolicy:
    ...
    syncOptions:
      - RespectIgnoreDifferences=true
    ...
  ignoreDifferences:
    - group: "*"
      kind: "Application"
      namespace: "*"
      jsonPointers:

>         # To allow manually turn off auto sync for apps and preserve that setting
>         - /spec/syncPolicy/automated
>         # These are automatically updated on a regular basis. Not ignoring last applied configuration since it's used for computing diffs after normalization.
>         - /metadata/annotations/argocd.argoproj.io~1refresh
>         - /operation

Reviewer 2: ```suggestion
        # Allow manually disabling auto sync for apps, useful for debugging.
        - /spec/syncPolicy/automated
        # These are automatically updated on a regular basis. Not ignoring last applied configuration since it's used for computing diffs after normalization.
        - /metadata/annotations/argocd.argoproj.io~1refresh
        - /operation
```

Reviewer 2: @andrii-korotkov-verkada Can you explain why you ignore the annotations? I wouldn't think this would add anything.

Author: @todaywasawesome, I've noticed that some annotations and /operation were constantly triggering out of sync status for the app of apps despite no changes in the manifests, since they could be added/removed/updated by automation. I thought it'll be cleaner to ignore them.
